grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-grid.directive.ts

```diff
@@ -0,0 +1,40 @@
+import { CdkDropList, moveItemInArray } from '@angular/cdk/drag-drop';
+import { Directive, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+@Directive({
+  selector: 'clr-datagrid[customClrColumnOrderingGrid]',
+})
+export class CustomClrColumnOrderingGridDirective implements OnInit, OnDestroy {
+  @Input() columns!: any[];
+  @Output() columnsChange = new EventEmitter<any[]>();
+
+  grabbedColumn: any;
+  private subscription: Subscription | undefined;
+
+  constructor(private readonly cdkDropList: CdkDropList) {}
+
+  ngOnInit(): void {
+    this.cdkDropList.orientation = 'horizontal';
+    this.subscription = this.cdkDropList.dropped
+      .pipe(
+        tap(event => {
+          this.reorderColumn(event);
+        })
+      )
+      .subscribe();
+  }
+
+  ngOnDestroy(): void {
+    this.subscription?.unsubscribe();
+  }
+
+  reorderColumn(event: { previousIndex: number; currentIndex: number }) {
+    if (event.currentIndex !== event.previousIndex) {
+      const value = [...this.columns];
+      moveItemInArray(value, event.previousIndex, event.currentIndex);
```

Comment:
I noticed when I used this function in my implementation, that if I was moving the column to the right (to later in the array) I needed to subtract 1 from the target index since everything after the previousIndex will shift left. I haven't pulled down your branch, but you may have the same issue

Reply:
My old implementation was indeed broken. I believe this one works as expected. I've tested it and it seems to work.

Reply:
Looks like the Angular cdk `dropped` event handles this case for us, unlike the `cdsDraggableChange` event.